### PR TITLE
Add health check and development script

### LINKED ISSUE #49: Health endpoint
For people deploying behind a load balancer, a simple healthcheck endpoint that verifies connection to the database would be very helpful

### PR DESCRIPTION
**Pull request by will2hew**
Adds a simple health check endpoint that verifies the connection to the database. 

This also contains a development server script I personally find helpful, but am happy to remove if you'd prefer. 

Closes #49

### CODE REVIEW

File: apps/server/src/integrations/health/health.controller.ts
Comment (Philipinho) [inline apps/server/src/integrations/health/health.controller.ts:1]: Looks good. We need Redis check too. 
Redis is linked with queues, collaboration and socket.io.

You can use the logger to log the health check on error.

File: package.json [hunk 1]
```diff
@@ -12,7 +12,8 @@
     "client:dev": "nx run client:dev",
     "server:dev": "nx run server:start:dev",
     "server:start": "nx run server:start:prod",
-    "email:dev": "nx run @docmost/transactional:dev"
+    "email:dev": "nx run @docmost/transactional:dev",
+    "dev": "pnpm concurrently -n \"frontend,backend\" -c \"cyan,green\" \"pnpm run client:dev\" \"pnpm run server:dev\""
```
Comment (Philipinho) [inline package.json:16]: LGTM.

File: apps/server/src/integrations/health/health.controller.ts
Comment (will2hew) [reply apps/server/src/integrations/health/health.controller.ts:1]: Sounds good, done

**Comment by will2hew** [comment]
@Philipinho I have added the requested redis check and logging. 

I would note, that currently the entire application crashes when Postgres and/or Redis connections are lost, so this health check does perform its job - we should probably make the app fail in a more graceful way. I think that should be a follow up to this however.

File: apps/server/src/integrations/health/health.controller.ts [hunk 2]
```diff
@@ -5,22 +5,33 @@ import {
...
   @HttpCode(HttpStatus.OK)
   async health() {
     try {
+      const redis = new Redis(this.environmentService.getRedisUrl());
+
       await sql`SELECT 1=1`.execute(this.db);
+      await redis.ping();
     } catch (error) {
-      throw new InternalServerErrorException('Health check failed');
+      this.logger.error('Health check failed');
```
Comment (Philipinho) [inline apps/server/src/integrations/health/health.controller.ts:33]: One last thing. We should add the `error` object to the log so we can know what went wrong.

**Comment by Philipinho** [comment]
> that currently the entire application crashes when Postgres and/or Redis connections are lost
It only crashes when the database connection fails during startup after 10 retries. Once the app is running, lost db connections do not crash it.

A lost Redis connection will not crash it, both during startup and while running. Redis retries indefinitely.

**Comment by Philipinho** [comment]
I could not add my commits here. I created a new branch #64. 
I will merge the new branch after you review it (If you've got time).
Thank you for your contribution.

**Pull request closed without merging**